web3/utils/abi.py

```diff
@@ -388,15 +393,13 @@ def abi_to_signature(abi):
     return function_signature
 
 
+@curry
```

Comment:
I ❤️ this function soooo much.  Makes things that you end up `partial`-ing so much cleaner and elegant.